android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt

```diff
@@ -399,11 +407,24 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
             }
         }
 
+        val offlineBehaviorParam = if (amount > offlineModeTransactionLimit || (terminal.offlineStatus.sdk.offlinePaymentAmountsByCurrency[currency]?.toInt()
+                ?: 0) > offlineModeStoredTransactionLimit) { OfflineBehavior.REQUIRE_ONLINE }
+            else {
```

Comment:
same here, we don't need the `if` statement here we can just use the mapping below directly. The integration will keep track of the transaction limit and whether they have hit their own limit